### PR TITLE
Ldap fallthrough

### PR DESCRIPTION
**Pull request by csmithATsquiz**
Add an option to the ldap:LDAPMulti auth type so instead of having to specify the organization a user is in (or trying to work it out), let it try each organization in turn.

### CODE REVIEW

**Comment by csmithATsquiz** [comment]
Similar to PR 128.

**Comment by jaimeperez** [comment]
Hi @csmithATsquiz, thanks for your pull request!

A quick question regarding this fallthrough option. What would happen if a user has the same user account in two or more LDAP servers? Let's say I'm a student while I'm working at my university, and I have an account named "jaime" both in the students directory and the staff one, with the same password. Would it still be possible for me to select which LDAP I want to authenticate against? From a quick look at your code, it seems to me that wouldn't be possible, since the realm part of the username would be ignored and the first server being able to authenticate me would be the only one I could use.

**Comment by csmithATsquiz** [comment]
Hi @jaimeperez,
You're right - it would use the first one it finds (depending on the order you have the ldap trees specified).

In my particular case that's ok (this option is because I don't want the user to have to select their realm either with part of the username or by choosing from a dropdown), for others where they have the same account in multiple realms, it wouldn't work.

File: modules/ldap/lib/Auth/Source/LDAPMulti.php [hunk 1]
```diff
@@ -118,4 +122,23 @@ protected function getOrganizations() {
...
+	 * @param string $password  The password the user wrote.
+	 * @return array  Associative array with the users attributes.
+	 */
+	private function loginAllOrgs($username, $password, array $sasl_args = NULL) {
+		foreach ($this->orgs as $org => $orgDescription) {
+			try {
+				$userdetails = $this->ldapOrgs[$org]->login($username, $password, $sasl_args);
+				return $userdetails;
+			} catch (SimpleSAML_Error_Error $e) {
+				// User didn't exist in that organization, so keep going.
```
[the remark this answers is not in the document]
Comment (csmithATsquiz) [reply modules/ldap/lib/Auth/Source/LDAPMulti.php:138]: Probably. Given the original MR is 6 years old, let's just throw this out.

(We ran this code in prod for many years without any issues before moving to something else).

**Comment by csmithATsquiz** [comment]
Closing as this is 6 years old, and no longer relevant.

**Pull request closed without merging**